File: apps/sdkwork-appstore-mini-program/src/bootstrap/runtime.ts
import { seedRuntimeEnvFromBundle } from "./runtimeBundle";
import { createSdkClients } from "./sdkClients";
import { registerHostAdapters } from "./hostAdapters";
import { createRoutes } from "./routes";

/**
 * Mini program bootstrap: environment selection, host adapters, SDK client
 * construction, route assembly. Mirrors the Flutter and Harmony roots.
 *
 * `appApiBaseUrl` is forwarded into the runtime-env resolution, so a host that
 * already resolved a gateway origin can override the bundled profile value
 * without bypassing `resolveBaseUrl` (`ENVIRONMENT_SPEC.md` §6.3).
 */
export function bootstrapAppstoreMiniProgram(options: {
  appApiBaseUrl?: string;
  accessToken?: string;
} = {}) {
  seedRuntimeEnvFromBundle({ appApiBaseUrl: options.appApiBaseUrl });
  registerHostAdapters();
  const sdkClients = createSdkClients(options.accessToken);
  const routes = createRoutes();
  return { sdkClients, routes };
}

File: apps/sdkwork-appstore-flutter-mobile/lib/bootstrap/runtime.dart
import 'package:sdkwork_appstore_flutter_mobile_core/sdkwork_appstore_flutter_mobile_core.dart';

import 'host_adapters.dart';
import 'iam_runtime.dart';
import 'routes.dart';
import 'sdk_clients.dart';

/// Flutter mobile composition root.
class AppstoreMobileRuntime {
  const AppstoreMobileRuntime({
    required this.sdkClients,
    required this.routes,
  });

  final AppstoreAppSdkClients sdkClients;
  final List<SdkworkUiRouteContribution> routes;
}

Future<AppstoreMobileRuntime> bootstrap() async {
  createIamRuntime();
  registerHostAdapters();
  final sdkClients = createSdkClients();
  final routes = createRoutes();
  return AppstoreMobileRuntime(sdkClients: sdkClients, routes: routes);
}
